test/integration/controlplane/kube_apiserver_test.go

```diff
@@ -440,19 +440,22 @@ func testReconcilersAPIServerLease(t *testing.T, leaseCount int, apiServerCount
 
 	instanceOptions := kubeapiservertesting.NewDefaultTestServerOptions()
 
+	lock := sync.Mutex{}
```

Comment:
I can solve the flake problem by using locks. But I am not sure if it still maintains the original intention of the test. This problem should be a problem of concurrent duplicate registration....

Reply:
I'm having some trouble reproducing it but the fix makes sense. Looks at the semantics, it seems like we were doing things in parallel for performance rather than for something specific we needed to test with concurrency https://github.com/kubernetes/kubernetes/pull/93721

Reply:
I looked at the code and it seems that this is an error caused by always using the same name when registering, so I changed the suffix to randomly generate the name to fix this flake. 🤔 

https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-apiserver/app/testing/testserver.go#L207-L209

Reply:
This fix may not completely solve the problem, but it seems to be a quick and easy fix. Avoid this duplicate registration problem.

Reply:
We shouldn't change the component name for registering since it is used for determining feature gates. The problem seems to be related to https://github.com/kubernetes/kubernetes/issues/128163, where two apiservers cannot be started simultaneously because global state is shared.

Reply:
After taking a look, this change seems to be wrong...

Reply:
After looking at the issue, it seems that the changes to solve this problem are relatively large. . . 🤔

Reply:
If we want a quick fix, it seems like using lock is a more appropriate option?

Reply:
Using a lock kind of defeats the purpose of the parallelization. Could just call the functions in serial order without a goroutine?

Reply:
>If we want a quick fix, it seems like using lock is a more appropriate option?

why do we want a quick fix?

Reply:
At present, the original concurrency is modified, and it is changed to serial. The reason is: it seems that it cannot be executed in parallel when registering https://github.com/kubernetes/kubernetes/issues/128163.

I'm not sure if this is what we want

Reply:
@Jefftree I'm trying to understand the situation, is serializing the test defeating the purpose of the test of the purpose of try to optimize it for parallelizing?

For the later we should merge until https://github.com/kubernetes/kubernetes/issues/128163 is fixed